python/dgl/data/chem/utils.py

```diff
@@ -10,42 +12,583 @@
+    """One hot encoding for whether the atom is aromatic.
+
+    Parameters
+    ----------
+    atom : rdkit.Chem.rdchem.Atom
+        RDKit atom instance.
+    allowable_set : list of bool
+        Conditions to consider. Default: ``False`` and ``True``.
+    encode_unknown : bool
+        If True, map inputs not in the allowable set to the
+        additional last element. (Default: False)
+
+    Returns
+    -------
+    list
+        List of boolean values where at most one value is True.
+    """
+    if allowable_set is None:
+        allowable_set = [False, True]
+    return one_hot_encoding(atom.GetIsAromatic(), allowable_set, encode_unknown)
+
+
+def atom_is_aromatic(atom):
+    """Get whether the atom is aromatic.
+
+    Parameters
+    ----------
+    atom : rdkit.Chem.rdchem.Atom
+        RDKit atom instance.
+
+    Returns
+    -------
+    list
+        List containing one bool only.
+    """
+    return [atom.GetIsAromatic()]
+
+
+def atom_chiral_tag_one_hot(atom, allowable_set=None, encode_unknown=False):
+    """One hot encoding for the chiral tag of an atom.
+
+    Parameters
+    ----------
+    atom : rdkit.Chem.rdchem.Atom
+        RDKit atom instance.
+    allowable_set : list of rdkit.Chem.rdchem.ChiralType
+        Chiral tags to consider. Default: ``rdkit.Chem.rdchem.ChiralType.CHI_UNSPECIFIED``,
+        ``rdkit.Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CW``,
+        ``rdkit.Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CCW``,
+        ``rdkit.Chem.rdchem.ChiralType.CHI_OTHER``.
+    """
+    if allowable_set is None:
+        allowable_set = [Chem.rdchem.ChiralType.CHI_UNSPECIFIED,
+                         Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CW,
+                         Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CCW,
+                         Chem.rdchem.ChiralType.CHI_OTHER]
+    return one_hot_encoding(atom.GetChiralTag(), allowable_set, encode_unknown)
+
+
+def atom_mass(atom, coef=0.01):
+    """Get the mass of an atom and scale it.
+
+    Parameters
+    ----------
+    atom : rdkit.Chem.rdchem.Atom
+        RDKit atom instance.
+    coef : float
+        The mass will be multiplied by ``coef``.
+
+    Returns
+    -------
+    list
+        List containing one float only.
+    """
+    return [atom.GetMass() * coef]
+
+
+class ConcatFeaturizer(object):
+    """Concatenate the evaluation results of multiple functions as a single feature.
+
+    Parameters
+    ----------
+    data_type
+        The type of objects that we want to featurize. E.g. ``rdkit.Chem.rdchem.Atom``
+        or ``rdkit.Chem.rdchem.Bond``.
+    func_list : list
+        List of functions for computing molecular descriptors from objects of type
+        ``data_type``. Each function is of signature
+        ``func(data_type) -> list of float or bool or int``. The resulting order of
+        the features will follow that of the functions in the list.
+    """
+
+    def __init__(self, data_type, func_list):
```

Comment:
Strictly speaking, this can be used for any type of the data like atom, bond or molecule. To declare the type of the target for featurization is just for a data type checking during featurization.

Reply:
It's user's duty to ensure the function and the variable's type are matched in Python. For dynamic language, this is a bit weird. Another reason is `rdkit.Chem.rdchem.Atom` is too long to memorize. It's not ideal for user to pass this in.

Reply:
Done.